[mkdocs] feat: add new docs site

segfaultxavi opened the pull request:
The symbol-docs repo is outdated, and each SDK
has its documentation built and published independently.
This PR creates a new documentation site which includes:

- Textbook: descriptions of the technology behind Symbol. Meant to be referenced from the other books.
- User manual: instructions for users of our products, without code. So far: Desktop wallet and Shoestring.
- Developer manual: 50 tutorials with code and explanations, plus reference pages for the Python and JS SDKs, REST API, Serialization structures, WebSocket, and the Catapult configuration files.

All pages are available both in English and Japanese.

segfaultxavi committed:
[mkdocs] fix: update swagger dep

This fixes the link to download the OpenAPI spec.

segfaultxavi committed:
[mkdocs] feat: add helper script to deploy docs to gh-pages

EXTREMELY MANUAL
NO ERROR CHECKING
USE WITH CAUTION

zero4862 committed:
[mkdocs] feat: add monitoring status tutorial (#1737)

* [mkdocs] feat: add monitoring status page

* [mkdocs] fix: refactor js code

* [mkdocs] fix: lint files

* [mkdocs] fix: lint files

* [mkdocs] fix: apply direct suggestions from code review

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

* [mkdocs] fix: address feedback

* [mkdocs] fix: indentation

* [mkdocs] fix: long line

* [mkdocs] fix: address comments

* [mkdocs] fix: trigger jenkins

* [mkdocs] fix: address comments

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] fix: apply class renames correctly

They were only applied to methods.
This is required because class `SymbolTransactionFactory` is called
`TransactionFactory` in Python.
More can be added as they are discovered via the `class-remaps` key in the yml config file.

segfaultxavi committed:
[mkdocs] fix: update output log of messages tutorial

Also, added an extra blank line for readability.

zero4862 committed:
[mkdocs] feat: send messages tutorial (#1738)

* [mkdocs] feat: add send messages  page

* [mkdocs] fix: add links between tutorials

* [mkdocs] fix: add prerequisite

* [mkdocs] fix: lint files

* [mkdocs] fix: lint files

* [mkdocs] fix: lint files

* [mkdocs] fix: update links

* [mkdocs] fix: address conventions

* [mkdocs] fix: address conventions

* [mkdocs] fix: address conventions

* [mkdocs] fix: address conventions

* [mkdocs] fix: address comments

* [mkdocs] fix: address comments

* [mkdocs] docs: address comments

* [mkdocs] docs: address comments

zero4862 committed:
[mkdocs] feat: add create account from private key tutorial (#1739)

* [mkdocs] feat: add create from private key page

* [mkdocs] fix: line lenght

* [mkdocs] fix: spaces

* [mkdocs] docs: address comments

[mkdocs] docs: address comments

[mkdocs] docs: address comments

[mkdocs] docs: address comments

[mkdocs] docs: address comments

[mkdocs] docs: address comments

[mkdocs] docs: address comments

segfaultxavi committed:
[mkdocs] fix: add glossary links to accessors

44uk committed:
[mkdocs] feat: keep viewing path when switch language. (#1750)

[mkdocs] feat: add lang switcher overrides.

segfaultxavi committed:
[mkdocs] fix: dynamic links when class or method has numbers

zero4862 committed:
[mkdocs] docs: add query balance tutorial (#1751)

* [mkdocs] docs: add query balance

* [mkdocs] fix: apply suggestions from code review

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

* [mkdocs] fix: address comments

* [mkdocs] fix: address comments

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: add create from mnemonic tutorial (#1752)

* [mkdocs] docs: add create from mnemonic tutorial

[mkdocs] docs: change passphrase to password

[mkdocs] docs: improve warning

* [mkdocs] fix: address comments

[mkdocs] fix: address comments

segfaultxavi committed:
[mkdocs] feat: initial Textbook Japanese translation

zero4862 committed:
[mkdocs] docs: add faucet tutorial (#1759)

* [mkdocs] docs: add faucet tutorial

Adds "Getting Testnet Funds from the Faucet" tutorial and review prerequisites and next steps for all published tutorials to provide a better navigation.

* [mkdocs] fix: typo
* [mkdocs] fix: typo
* [mkdocs] fix: apply suggestions from code review
* [mkdocs] fix: address comments
* [mkdocs] fix: address comments

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] fix: correct amount to participate in finalization

segfaultxavi committed:
[mkdocs] fix: correct amount to participate in finalization [ja]

segfaultxavi committed:
[mkdocs] fix: document required Symbol SDK version

segfaultxavi committed:
[mkdocs] fix: clarify cosignature definition

segfaultxavi committed:
[mkdocs] fix: document missing GraphViz dependency

segfaultxavi committed:
[mkdocs] fix: add .venv to .gitignore

zero4862 committed:
[mkdocs] docs: add complete aggregate tutorial (#1772)

* [mkdocs] docs: add complete aggregate tutorial
* [mkdocs] fix: spaces
* [mkdocs] fix: review lines length
* [mkdocs] docs: address comments
* [mkdocs] fix: address comments

segfaultxavi committed:
[mkdocs] fix: Update ezglossary plugin and its patch

- The new version improves handling of HTML tags in term definitions.
- But its refuses to fix recursive definitions, so the patch  is
  still needed and needed to be adapted to the new version.
- Also, the regex has been fixed because it didn't cover cases like
  <p>Text <term:|label></p>

zero4862 committed:
[mkdocs] docs: add bonded aggregate tutorial (#1773)

segfaultxavi committed:
[mkdocs] fix: correct harvesting beneficiary percentage

segfaultxavi committed:
[mkdocs] feat: define mainnet and testnet glossary terms

And link to them from the rest of pages.

zero4862 committed:
[mkdocs] docs: update testnet endpoint (#1804)

segfaultxavi committed:
[mkdocs] fix: reference links for global functions

This is more a hack than a fix, but handling methods that can
live outside any class, both in Python and JavaScript, would
require a lot of refactor.

segfaultxavi committed:
[mkdocs] fix: make clear only one account can benefit from harvesting

daoka committed:
[mkdocs] fix: Japanese translation (#1826)

segfaultxavi committed:
[mkdocs] fix: Japanese glossary mismatch

segfaultxavi committed:
[mkdocs] fix: repeated sentence

zero4862 committed:
[mkdocs] docs: add account metadata tutorial (#1803)

* [mkdocs] docs: add account metadata tutorial
fix: update reference node
chore: edits
chore: edits
* [mkdocs] docs: apply review comments & rebase
* fix: address comments

segfaultxavi committed:
[mkdocs] fix: Sentence case

segfaultxavi committed:
[mkdocs] fix: make short title consistent with rest of section

All other short titles in the Accounts section start with a verb.

zero4862 committed:
[mkdocs] docs: add namespace tutorials (#1810)

* [mkdocs] docs: register namespace tutorials
* [mkdocs] fix: links
* fix: apply feedback
* fix: apply feedback
---------
Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] feat: new style for landing page cards

segfaultxavi committed:
[mkdocs] add: multisig tutorial (#1840)

* [mkdocs] add: initial working source code
* [mkdocs] add: multisig tutorial WIP
* [mkdocs] add: multisig tutorial
* [mkdocs] add: multisig tutorial (note on multivel)

segfaultxavi committed:
[mkdocs] add: tutorial about signing multisig transactions (#1844)

[mkdocs] add: multisig signing tutorial

segfaultxavi committed:
[mkdocs] add: allow tutorials that do not show the full code

segfaultxavi committed:
[mkdocs] fix: remove link to non-existing anchor

segfaultxavi committed:
[mkdocs] add: fee sponsorship tutorial (#1847)

zero4862 committed:
[mkdocs] docs: add link namespace tutorials (#1838)

* [mkdocs] docs: add link namespace tutorials
* fix: replace generateNamespaceId for generateNamespacePath
* fix: use namespace generate id for mosaics
* chore: apply feedback
* fix: add note on unique subnamespace ids
* chore: rebase and link to alias guides from previous tutorials
* [mkdocs] fix: internal formatting

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: create mosaic tutorials (#1855)

* [mkdocs] docs: create mosaic tutorials
* [mkdocs] fix: use "lease fee" consistently
Instead of "rental fee", even though that's the underlying name in the code and REST endpoints.
* [mkdocs] review feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] fix: do not highlight diagram nodes on hover if there is no link

zero4862 committed:
[mkdocs] add namespace and mosaic metadata tutorials (#1854)

* [mkdocs] add namespace and mosaic metadata tutorials
* [mkdocs] reviewer feedback
* [mkdocs] reviewer feedback
* [mkdocs] reviewer feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] add: metadata page for the textbook

segfaultxavi committed:
[mkdocs] add: Account restrictions tutorial (#1863)

* [mkdocs] fix: clarify restrictions in the textbook
* [mkdocs] add: account restrictions tutorial (WIP: py code)
* [mkdocs] add: JS code
* [mkdocs] add: Account restrictions tutorial

segfaultxavi committed:
[mkdocs] add: Devbook Reference Serialization tables

segfaultxavi committed:
[mkdocs] fix: remove links to old docs from Serialization tables

zero4862 committed:
[mkdocs] docs: add modify mosaic definition tutorial (#1858)

* [mkdocs] docs: add mosaic definition tutorial
* fix: address comments
* [mkdocs] reviewer feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: add resolve namespace from receipt tutorial (#1865)

* [mkdocs] docs: add resolve namespace from receipt tutorial
* [mkdocs] docs: address comments
* [mkdocs] reviewer feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] chore: review consistency between tutorials (#1872)

* [mkdocs] review consistency
* [mkdocs] rollback edit
* [mkdocs] fix: formatting

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] fix: add source address back to metadata tutorials (#1874)

* [mkdocs] fix: add source address back to metadata tutorials
* [mkdocs] fix: add source address back to metadata tutorials

zero4862 committed:
[mkdocs] docs: revoke mosaic tutorial (#1873)

* [mkdocs] docs: revoke mosaic tutorial
* [mkdocs] fix: address comments
* [mkdocs] fix: address comments

zero4862 committed:
[mkdocs] docs: add chain and finalization height tutorial (#1875)

segfaultxavi committed:
[mkdocs] add: WebSockets reference guide

zero4862 committed:
[mkdocs] docs: add prove transaction tutorial (#1889)

* [mkdocs] docs: add prove transaction tutorial
* [mkdocs] fix: address comments
* [mkdocs] typos

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: add listen new blocks tutorial (#1891)

* [mkdocs] docs: add listen new blocks tutorial
* [mkdocs] fix: address comments

segfaultxavi committed:
[mkdocs] fix: missing https in URL

segfaultxavi committed:
[mkdocs] add: mosaic restriction tutorial (#1890)

* [mkdocs] add: mosaic restriction tutorial py code
* [mkdocs] add: mosaic restrictions JS code

And fixes to the Python code.

* [mkdocs] add: Mosaic Restrictions tutorial

segfaultxavi committed:
[mkdocs] feat: allow links with preview to serialization tables

Using <ser:Mosaic> you get a link to the appropriate section of the table,
with a description on hover and proper code formatting.

Warning: use versioned names (i.e. add `V1` when required).

segfaultxavi committed:
[mkdocs] add: links to serialization table

zero4862 committed:
[mkdocs] docs: add listen to transaction flow tutorial (#1892)

* [mkdocs] docs: add listen to transaction flow tutorial
[mkdocs] docs: review tutorial
[mkdocs] docs: review tutorial
[mkdocs] docs: review tutorial
[mkdocs] docs: add tutorial to sidebar
[mkdocs] docs: add tutorial to sidebar
* [mkdocs] docs: fix websocket links
* [mkdocs] docs: address comments
[mkdocs] docs: address comments
* [mkdocs] review

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] remove Java SDK from docs

Work on this SDK is paused and probably won't be ready for the release of the new docs.

Reference pages are still being generated, but it does not take long (compared to JS).

segfaultxavi committed:
[mkdocs] silence some annoying log messages

segfaultxavi committed:
[mkdocs] cleanup Python reference pages

Filtered out:
- NEM pages.
- empty pages.

The new config in mkdocs.base.yml allows to easily filter-out Symbol pages from the NEM docs too.

zero4862 committed:
[mkdocs] docs: add listen transaction error tutorial (#1898)

zero4862 committed:
[mkdocs] docs:  listening to bonded tx flow tutorial (#1899)

* [mkdocs] docs:  listening to bonded tx flow tutorial
[mkdocs] docs:  listening to bonded tx flow tutorial
* [mkdocs] review

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] fix: low contrast links in devbook

Normal link color in light mode should be the dark color, lightening on hover.
Dark mode is the opposite.

segfaultxavi committed:
[mkdocs] fix: invisble dylinks on first visit

Dylinks show links to Python or JS reference, depending on the selected language in the code tabs.
But upon first visit, when no language has been selected, the dylinks were empty.

This fix selects Python as the starting language, matching the default selected tab.

segfaultxavi committed:
[mkdocs] add: footer link to old docs site

And link titles.

segfaultxavi committed:
[mkdocs] fix: Japanese footer link titles

segfaultxavi committed:
[mkdocs] fix: remove non-markdown tags from reference guides

segfaultxavi committed:
[mkdocs] fix: use new docs URL in OpenAPI spec

It will be changed again when we deploy the new docs to their final URL.

segfaultxavi committed:
[mkdocs] fix: indentation of config YAML

segfaultxavi committed:
[mkdocs] add: scams and security page + redirect

This was an old page we had, rewritten and placed in the userbook.
Plus a redirect so links to the old page are taken to the new page.

segfaultxavi committed:
[mkdocs] fix: serialization links to entities

Glossary links using <ser:> worked fine,
but the table itself has hardcoded links to
#anchors that were not present.

zero4862 committed:
[mkdocs] docs: add query currency supply tutorial (#1907)

* [mkdocs] docs: add query currency supply tutorial

* Update mkdocs/pages/en/devbook/network-currency/query-currency-supply.md

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

* [mkdocs] update to latest OpenAPI spec

zero4862 committed:
[mkdocs] docs: query block rewards (#1913)

* [mkdocs] docs: query block rewards
* [mkdocs] docs: query block rewards

segfaultxavi committed:
[mkdocs] add: multilevel multisig textbook diagram

zero4862 committed:
docs: add prove mosaic definition tutorial (#1906)

* docs: add prove mosaic state tutorial
docs: add prove mosaic state tutorial
* docs: add prove mosaic state tutorial
* [mkdocs] docs: address comments

zero4862 committed:
[mkdocs] refactor: use SDK Address methods for hex-to-address conversion (#1914)

[mkdocs] docs: use SDK Address methods for hex-to-address conversion

Spotted this improvement while working in our of the latest tutorials.

Replaces manual `bytes.fromhex/Buffer.from` patterns with the SDK built-in methods `Address.from_decoded_address_hex_string` (Python) and `Address.fromDecodedAddressHexString` (JS) across namespace tutorials.

zero4862 committed:
[mkdocs] docs: add get mosaic and namespace definitions tutorials (#1916)

* [mkdocs] docs: add get mosaic and namespace definitions tutorials
* [mkdocs] docs: address comments
* [mkdocs] docs: address comments

zero4862 committed:
[mkdocs] docs: transaction batching (#1917)

* [mkdocs] docs: transaction batching
* [mkdocs] docs: transaction batching
* [mkdocs] review feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] add: Network Time tutorial and textbook entry

segfaultxavi committed:
[mkdocs] fix: get private key from environment in Transfer tutorial

segfaultxavi committed:
[mkdocs] fix: typo in README

segfaultxavi committed:
[mkdocs] fix: progress plots generator

Archived tasks were not taken into account.

zero4862 committed:
[mkdocs] docs: use Address.fromNamespaceId SDK method in link-namespace-to-address tutorial (#1940)

segfaultxavi committed:
[mkdocs] add: warning about exporting profiles from the wallet

segfaultxavi committed:
[mkdocs] fix: update OpenAPI spec to latest version

segfaultxavi committed:
[mkdocs] feat: replace Swagger with Scalar

This is a more modern OpenAPI renderer.
It is still client-side, so we still get the mkdocs warnings about missing anchors,
but at least we can get rid of the hacky scroll-to-endpoint script.

segfaultxavi committed:
[mkdocs] fix: prevent reentrancy in Scalar OpenAPI renderer

segfaultxavi committed:
[mkdocs] fix: do not initialize Scalar until page is loaded

This gets rid of race conditions. Hopefully.

segfaultxavi committed:
[mkdocs] fix: cleanup empty reference pages

Added them to the ignore list.

segfaultxavi committed:
[mkdocs] fix: silence invalid warnings about missing anchors in REST page

The page is dynamic, the anchors exist at runtime but mkdocs does not know about them.

segfaultxavi committed:
[mkdocs] fix: missing import in Transfer Python tutorial

Which I broke in 3964edc2c5256ef6417c8c779503b66f33e07d0d

segfaultxavi committed:
[mkdocs] update: Symbol SDK version to 3.3.1

zero4862 committed:
[mkdocs] docs: add atomic swaps tutorial (#1938)

daoka committed:
[mkdocs] docs: translate the Japanese dev book and user guide (#1941)

* [mkdocs] fix: Japanese translation
* [mkdocs] add: devbook Japanese translation
* [mkdocs] fix: devbook translation
* [mkdocs] fix: devbook translation
* [mkdocs] fix textbook lacks contents
* [mkdocs] fix devbook translate
* [mkdocs] fix devbook translate
* [mkdocs] fix devbook translate
* [mkdocs] fix: missing anchor link
* [mkdocs] fix translate terminology

fix: 'Divisibility' translate from 分割性(bunkatsusei) to 可分性(kabunsei)
because follow established practice

* [mkdocs] fix: missing links
* [mkdocs] fix: missing page

added devbook/namespaces/get-namespace-info (ja)

* [mkdocs] fix: missing links
* [mkdocs] add: userbook translate
* [mkdocs] fix: userbook translate
* [mkdocs] fix: Userbook translate
* [mkdocs] add: warning about exporting profiles translate
* [mkdocs] fix: devbook translate
* [mkdocs] fix: devbook translate
* [mkdocs] fix: translate
* [mkdocs] fix: devbook's anchor links
* [mkdocs] fix: textbook's anchor links
* [mkdocs] fix: missing menu
* [mkdocs] fix: userguide links
* [mkdocs] fix: remove manual anchors from page titles

They are unused, and they show in the TOC.

* [mkdocs] fix: link to Japanese textbook for "signature"
* [mkdocs] fix: merge mail accounts for Daoka
* [mkdocs] fix: Add REST and remove Java from Japanese reference pages
* [mkdocs] add WebSockets translation
* [mkdocs] fix: Websokect translation
* [mkdocs] add: atomic-sawp translation
* [mkdocs] fix: transfer translation
* [mkdocs] fix: missing translation
* [mkdocs] fix: move faucet images to their proper place

The overrides directory exists precisely to share assets between languages.

* [mkdocs] fix: final formatting pass

Remove blank lines, trailing whitespace, and make the linter happy.

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: add cross chain swap tutorial and concept (#1925)

* [mkdocs] docs: add cross chain swap tutorial and concept
* [mkdocs] docs: review cross-chain concept
* [mkdocs] docs: review cross-chain tutorial
* [mkdocs] reviewer feedback

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

zero4862 committed:
[mkdocs] docs: update sdk 3.3.1 (#1955)

* [mkdocs] docs: update sdk 3.3.1
* [mkdocs] fix: polish japanese translation

---------

Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

daoka committed:
[mkdocs] docs: translate cross-chain swap (#1957)

* [mkdocs] add: cross chain's translation
* [mkdocs] fix: inconsistencies in Japanese text
* [mkdocs] fix: feedback

---------

Co-authored-by: daoka <daoka@symbolsyndicate.us>
Co-authored-by: Xavi Artigas <xavierartigas@yahoo.es>

segfaultxavi committed:
[mkdocs] add: shoestring guides (#1903)

* [mkdocs] add: shoestring guides (WIP)
* [mkdocs] add: clarify shoestring deps
* [mkdocs] add: shoestring guides (WIP)
* [mkdocs] add: Catapult properties reference guides
* [mkdocs] Shoestring guides WIP
* [mkdocs] add: Shoestring guides WIP
* [mkdocs] add: shoestring WIP
* [mkdocs] fix: reviewer feedback
* [mkdocs] Shoestring maintenance WIP
* [mkdocs] Shoestring guides WIP
* [mkdocs] fix: abort deploy-gh-pages script on error
* [mkdocs] shoestring WIP
* [mkdocs] shoestring migration notes

segfaultxavi committed:
[mkdocs] fix: links to nonexisting migration guide

segfaultxavi committed:
[mkdocs] fix: voting rewards in Shoestring guide

segfaultxavi committed:
[mkdocs] fix: assorted typos and missing information

Spotted while working on the NEM documentation.

segfaultxavi committed:
[mkdocs] fix: missing admonition title

segfaultxavi committed:
[mkdocs] fix: typos in Shoestring guides

daoka committed:
[mkdocs] add: translate shoestring (#1962)

segfaultxavi committed:
[mkdocs] feat: build OpenAPI spec before docs

segfaultxavi committed:
[mkdocs] fix: update package-lock.json

segfaultxavi committed:
[mkdocs] fix: anchor link in Japanese version

segfaultxavi committed:
[mkdocs] add: sentence before every full code block explaining the structure of the tutorials

The sentence is taken from the YAML config file so it can be translated.

segfaultxavi committed:
[mkdocs] fix: clarify mosaics with multiple aliases

segfaultxavi committed:
[mkdocs] fix: typo

segfaultxavi committed:
[mkdocs] fix: silence useless warning about files not in the navigation menu

The files are: 404.md, which obviously does not belong to the navigation menu,
and the Java index file, which is disabled for now.

segfaultxavi committed:
[mkdocs] feat: tag tutorials with their difficulty level

Set the tag in the tutorial's frontmatter, and it will show in the tutorial header,
and in an autogenerated table in the devbook welcome page.

The tag is `tutorial_level`, and the possible values are `beginner`, `intermediate`, and
`advanced`.

segfaultxavi committed:
[mkdocs] fix: proper path separator on Windows

segfaultxavi committed:
[mkdocs] fix: fully disable building Java docs

segfaultxavi committed:
[mkdocs] fix: sort tutorials in summary table

In the same order they are in the navigation.

segfaultxavi committed:
[mkdocs] feat: tag tutorials with their difficulty level

segfaultxavi committed:
[mkdocs] fix: translate sections in tutorial summary table

segfaultxavi committed:
[mkdocs] feat: parse section tags from tutorial code

[>section_name] and [<section_name] tags can be added to tutorial source code comments,
and new macros code_full_tagged() and code_snippet_tagged() can be used to insert
line-number-independent snippets.

segfaultxavi committed:
[mkdocs] fix: remove saldom-used descriptor reference

We can bring it back later if needed.
Right now, it complicates the migration of snippets from line-numbers to section names.

segfaultxavi committed:
[mkdocs] fix: proper extraction of snippets

segfaultxavi committed:
[mkdocs] feat: use tags instead of line numbers in snippets

This allows moving code around without breaking the tutorials.

segfaultxavi committed:
[mkdocs] add: scripts for snippet migration

For reference only. The migration is complete and we should
not need them anymore.

segfaultxavi committed:
[mkdocs] fix: don't forget to make the linter happy

segfaultxavi committed:
[mkdocs] fix: some Japanese pages missed the conversion to named snippets

segfaultxavi committed:
[mkdocs] add: drop support for line-number-based snippets

segfaultxavi committed:
[mkdocs] fix: stop copying source code to build directory

The code and output files have been moved to a `snippets`
directory, because everything in `overrides` that does not look
to MkDocs like a static file is copied to the output build folder
and is therefore published in the site.

segfaultxavi committed:
[mkdocs] fix: python import order

segfaultxavi committed:
[mkdocs] fix: deprecation warning on Python 3.14

segfaultxavi committed:
[mkdocs] fix: feedback on the Textbook

segfaultxavi committed:
[mkdocs] fix: deprecation warning with Python 3.14

segfaultxavi committed:
[mkdocs] add: SDK installation troubleshooting

segfaultxavi committed:
[mkdocs] fix: clarify min and median fees

segfaultxavi committed:
[mkdocs] fix: less-strict linting rules for tutorial code

We prioritize readability and compactness.

Wayonb committed:
[mkdocs]: add linting scripts

segfaultxavi committed:
[mkdocs] fix: first batch of linting fixes

segfaultxavi committed:
[mkdocs] fix: lint python tools

segfaultxavi committed:
[mkdocs] fix: pycodestyle linting fixes

segfaultxavi committed:
[mkdocs] fix: rename all tutorial code to use _ instead of -

As per the linting rules.

segfaultxavi committed:
[mkdocs] fix: pylint linting fixes

segfaultxavi committed:
[mkdocs] fix: eslint JS linting fixes

segfaultxavi committed:
[mkdocs] fix: lint error

We don't want to force installing ethers for every tutorial.

segfaultxavi committed:
[mkdocs] fix: eslint operator linebreak

segfaultxavi committed:
[mkdocs] fix: eslint mixed operators

segfaultxavi committed:
[mkdocs] fix: add Symbol SDK as a dep for the docs

It's not needed to build the docs, but it's needed to lint the sample code.

segfaultxavi committed:
[mkdocs] fix: pylint about uninstalled modules

We do not want to force installing these modules to build the docs.

segfaultxavi committed:
[mkdocs] fix: remove button background

Background appeared on the code copy buttons when updated the
Material plugin in 41b44b719e.

segfaultxavi committed:
[mkdocs] fix: remove unnecessary try/catch

segfaultxavi committed:
[mkdocs] fix: add links to WebSockets tutorials

segfaultxavi committed:
[mkdocs] fix: make py and js code consistent

segfaultxavi committed:
[mkdocs] fix: avoid signing twice (configure-multisig)

segfaultxavi committed:
[mkdocs] fix: use asterisks as list markers

Only a few files didn't use the rule.

segfaultxavi committed:
[mkdocs] fix: feedback on configure-multisig

segfaultxavi committed:
[mkdocs] fix: account-restrictions flag meaning

They have nothing to do with XOR.

segfaultxavi committed:
[mkdocs] fix: rename mult to multiplier

For extra clarity.

segfaultxavi committed:
[mkdos] fix: transaction max lifetime is 6h

segfaultxavi committed:
[mkdocs] fix: remove hardcoded cosignature size

Retrieve it from the SDK instead.

segfaultxavi committed:
[mkdocs] fix: use transaction.to_json() wherever possible

We requested this addition to the SDK to simplify one tutorial,
but it can be used in several other places.
We better be consistent.

segfaultxavi committed:
[mkdocs] fix: clarify that bonded agg txes do not happen fully on-chain

Technically, the partial transaction pool is not part of the chain state.

segfaultxavi committed:
[mkdocs] fix: clarify execution order in fee sponsorship

segfaultxavi committed:
[mkdocs] fix: be explicity about using a secure RNG in Python

os.urandom() probably already uses a secure random number
generator underneath, but let's be explicit about it.

segfaultxavi committed:
[mkdocs] fix: reorder Textbook pages and summary table

segfaultxavi committed:
[mkdocs] fix: reorganize tutorials

Use a more sensible order, and move sign-multisig from
accounts to transactions.

segfaultxavi committed:
[mkdocs] fix: capitalization of glossary terms

segfaultxavi committed:
[mkdocs] fix: add test tx to alias-address tutorial

segfaultxavi committed:
[mkdocs] fix: add test tx to alias-mosaic tutorial

segfaultxavi committed:
[mkdocs] fix: use the new isAlias method in resolve-namespace

segfaultxavi committed:
[mkdocs] fix: clarify that receips are ordered

And make use of it to bail out of the loop earlier.

segfaultxavi committed:
[mkdocs] fix: do not use abbreviations in query-currency-supply

segfaultxavi committed:
[mkdocs] fix: clarify finalization block hash usage

segfaultxavi committed:
[mkdocs] fix: clarify nemesis time format

segfaultxavi committed:
[mkdocs] fix: output line numbers

segfaultxavi committed:
[mkdocs] add: macro to allow dynamic variable names

Use `tutorial.var('snake_case_name')` and get a variable name that switches between snake_case and camelCase depending on the selected language.

segfaultxavi committed:
[mkdocs] fix: proper hiding of dynamic link input fields

segfaultxavi committed:
[mkdocs] fix: Japanese build

segfaultxavi committed:
[mkdocs] add: adapt names to selected language

Use snake_case or camelCase for variables and method
names depending on the selected language.

Also, use the <ser:> glossary links for all transaction descriptors.

segfaultxavi committed:
[mkdocs] fix: describe VALID_NEGATIVE proofs

segfaultxavi committed:
[mkdocs] fix: clarify state proof race condition

segfaultxavi committed:
[mkdocs] fix: add signature to block checks

segfaultxavi committed:
[mkdocs] fix: review feedback on ws-bonded-flow

segfaultxavi committed:
[mkdocs] fix: reviewer feedback of the cryptography pages

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on multiple textbook pages

segfaultxavi committed:
[mkdocs] fix: add numerical summary tables to textbook pages

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the blocks textbook page

segfaultxavi committed:
[mkdocs] fix: lots of tweaks to the Nodes textbook page

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the textbook

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the userbook

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the textbook

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the textbook

segfaultxavi committed:
[mkdocs] fix: remove unused section from serialization tables

segfaultxavi committed:
[mkdocs] fix: reviewer feedback on the textbook

segfaultxavi committed:
[mkdocs] fix: snippet boundary

cryptoBeliever committed:
[mkdocs] Fix typos and grammar in devbook, userbook, and README (#2007)

segfaultxavi committed:
[mkdocs] fix: use JPGs instead of PNGs for faucet screenshots

They are much smaller.

segfaultxavi committed:
[mkdocs] fix: consistency pass

Remove contractions and fix some grammar.

segfaultxavi committed:
[mkdocs] update: CONTRIBUTING instructions

segfaultxavi committed:
[mkdocs] fix: tweak tutorial complexity tags

zero4862 committed:
[mkdocs] docs: add error handling to query balance tutorial (#2014)

- Adds error handling to query balance tutorial (`try / catch`)
- Indents `Account holds N mosaics` code under `else` in Python to match JS implementation.

segfaultxavi committed:
[mkdocs] fix: check NODE_URL envvar in transfer tx tutorial

daoka committed:
[mkdocs] fix: refine Japanese Symbol documentation (#2022)

This update reflects the Japanese translation of the Symbol
documentation that was updated after the end of April.

---------

Co-authored-by: daoka <daoka@symbolsyndicate.us>
Co-authored-by: Xavi Artigas <xavi@symbolsyndicate.us>

zero4862 committed:
[mkdocs] docs: improve chain height tutorial

zero4862 committed:
[mkdocs] docs: improve monitoring transaction status tutorial (#2029)

Brings improvements from https://github.com/NemProject/nem/pull/890

---------

Co-authored-by: Xavi Artigas <xavi@symbolsyndicate.us>

segfaultxavi committed:
[mkdocs] fix: broken links in JA pages

segfaultxavi committed:
[mkdocs] fix: add missing note in Japanese version

zero4862 committed:
[mkdocs] docs: improve currency supply tutorial (#2046)

segfaultxavi committed:
[mkdocs] fix: Japanese translation of get-mosaic-info tutorial

Had a broken link among other issues.

zero4862 committed:
[mkdocs] docs: improve query block rewards tutorial (#2047)

segfaultxavi committed:
[mkdocs] fix: Japanese pages

* Broken links and anchors
* Standardize terminology
* Add missing sections
* Unify style with English version

segfaultxavi committed:
[mkdocs] fix: move complexity label tags under the title

segfaultxavi committed:
[mkdocs] fix: remove unnecessary manual anchors from Japanese titles

segfaultxavi committed:
[mkdocs] fix: remove tutorial complexity tag from setup page

segfaultxavi committed:
[mkdocs] fix: always provide title in metadata

Otherwise, it's taken from the page's H1 header,
which now includes the complexity level tag.

segfaultxavi committed:
[mkdocs] fix: update copyright year

segfaultxavi committed:
[mkdocs] fix: broken link to anchor

segfaultxavi committed:
[mkdocs] feat: optimize homepage loading time

Preload all images and strip metadata from PNGs.

segfaultxavi committed:
[mkdocs] feat: silence Material for MkDocs warning

It's a warning about the upcoming incompatible MkDocs 2
which we will not be using.

Jaguar0625 commented on docs/guides/account/scams-and-security.html:
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
```
what is purpose of this?

docs seems to just have two static html files now, any reason not to move them under mkdocs?

segfaultxavi replied:
Yes, this is a redirect, needed because the Desktop wallet has a link to this page.
It can't be under `mkdocs` because every page there will be published under `/en/` or `/ja/`.
The only doc pages I plan to have outside `mkdocs` are these redirects, and I need a few more.

Jaguar0625 replied:
it has a link to the github docs or the hosted docs? if the latter, maybe you could achieve the same with a post build step?

segfaultxavi replied:
All these files have been moved to `/mkdocs/templates/deploy` and are now copied to the output directory during build.

Jaguar0625 commented on linters/scripts/lint_indents.sh:
```diff
@@ -14,5 +14,7 @@ fi
 		--type-not=rst \
 		--type-not=yaml \
+		--type-not=jinja \
+		--type-not=log \
```
are you checking in log files?

Jaguar0625 replied:
ok, this is console output 👍

Jaguar0625 commented on mkdocs/cats_lexer/cats_lexer.py:
```diff
@@ -0,0 +1,22 @@
+	tokens = {
+		'root': [
+			(r'\b(import|using|enum|struct|make_const|make_reserved|binary_fixed|sizeof|if|equals|not equals|has|not has|array|__FILL__|__value__|abstract|inline|pad_last|not pad_last)\b', Keyword),
+			(r'\b(__FILL__|__value__)\b', Keyword.Reserved),
```
do you intend for these to be both Keyword and Keyword.Reserved?

segfaultxavi replied:
The color is not even different, so probably not 😅
I'll change it.

Jaguar0625 commented on mkdocs/cats_lexer/cats_lexer.py:
```diff
@@ -0,0 +1,22 @@
+
+	tokens = {
+		'root': [
+			(r'\b(import|using|enum|struct|make_const|make_reserved|binary_fixed|sizeof|if|equals|not equals|has|not has|array|__FILL__|__value__|abstract|inline|pad_last|not pad_last)\b', Keyword),
```
is there reason you didn't want not to be keyword?

Jaguar0625 replied:
missing a handful of things like `in` `not in`

segfaultxavi replied:
Where can I find the whole list?
I worked from the available CATS files rather than from source.

Jaguar0625 replied:
grammar is here: https://github.com/symbol/symbol/blob/8aad7e66ae70509f3498aa74566659db472979b9/catbuffer/parser/catparser/grammar/catbuffer.lark

segfaultxavi replied:
Reimplemented the lexer from the grammar file. That should take care of the rest of comments in this file.

Jaguar0625 commented on mkdocs/cats_lexer/cats_lexer.py:
```diff
@@ -0,0 +1,22 @@
+			(r'\b(__FILL__|__value__)\b', Keyword.Reserved),
+			(r'\b(int8|int16|int32|int64|uint8|uint16|uint32|uint64)\b', Keyword.Type),
+			(r'@(is_bitwise|is_aligned|is_size_implicit|size|initializes|discriminator|comparer|alignment|is_byte_constrained|sort_key|sizeref)\b', Keyword),
+			(r'(0x)?\d+', Number),
```
technically, hex numbers are supported (but i don't think we use in our schemas)

Jaguar0625 commented on mkdocs/cats_lexer/cats_lexer.py:
```diff
@@ -0,0 +1,22 @@
+			(r'\b(import|using|enum|struct|make_const|make_reserved|binary_fixed|sizeof|if|equals|not equals|has|not has|array|__FILL__|__value__|abstract|inline|pad_last|not pad_last)\b', Keyword),
+			(r'\b(__FILL__|__value__)\b', Keyword.Reserved),
+			(r'\b(int8|int16|int32|int64|uint8|uint16|uint32|uint64)\b', Keyword.Type),
+			(r'@(is_bitwise|is_aligned|is_size_implicit|size|initializes|discriminator|comparer|alignment|is_byte_constrained|sort_key|sizeref)\b', Keyword),
```
how did you split up keywords between first line and this one?

segfaultxavi replied:
These ones start with `@`.

Jaguar0625 commented on sdk/javascript/docs/typedoc.json:
```diff
@@ -1,16 +1,44 @@
-    "$schema": "https://typedoc.org/schema.json",
+    "$schema": "https://typedoc-plugin-markdown.org/schema.json",
 
     "entryPoints": [
```
are you sure about this change? everything should be accessible from indexes and that should hide private (non-exporeted) objects

Jaguar0625 replied:
if not clear, this is in reference to `entryPoints`

segfaultxavi replied:
If I only export the indexes:
```json
        "../ts/src/index.d.ts",
        "../ts/src/symbol/index.d.ts",
```
Then TypeDoc complains:
```
[warning] NetworkTimestampDatetimeConverter, defined in symbol-sdk/ts/src/NetworkTimestamp.d.ts, is referenced by symbol.Network.datetimeConverter but not included in the documentation
[warning] TreeNode, defined in symbol-sdk/ts/src/symbol/merkle.d.ts, is referenced by symbol.deserializePatriciaTreeNodes but not included in the documentation
[warning] Bip32Node, defined in symbol-sdk/ts/src/Bip32.d.ts, is referenced by index.Bip32.fromMnemonic but not included in the documentation
[warning] TryDecodeResult, defined in symbol-sdk/ts/src/symbol/MessageEncoder.d.ts, is referenced by symbol.MessageEncoder.tryDecode but not included in the documentation
[warning] NetworkTimestamp, defined in symbol-sdk/ts/src/NetworkTimestamp.d.ts, is referenced by symbol.NetworkTimestamp.addHours but not included in the documentation
[warning] MerklePart, defined in symbol-sdk/ts/src/symbol/merkle.d.ts, is referenced by symbol.proveMerkle.merklePath but not included in the documentation
[warning] Network, defined in symbol-sdk/ts/src/Network.d.ts, is referenced by index.NetworkLocator.findByIdentifier.TNetwork but not included in the documentation
```

Does this mean the index files are incomplete?

Jaguar0625 replied:
with just the indexes, it seems to not be documenting indirectly referenced objects, so i guess what you have originally is fine. Bip32Node, for example, can be returned to the caller by fromMnemonic, but it should not be created directly (which is why it is not in index).

Jaguar0625 commented on sdk/javascript/docs/typedoc.json:
```diff
@@ -1,16 +1,44 @@
+    "readme": "none",
+    "disableSources": true,
+    "gitRemote": "upstream",
+    "exclude": ["**/nem/*"],
```
do you intend to change this for nem docs?

segfaultxavi replied:
Yes, NEM is excluding `/symbol/` instead (but in the NEM repo this file is in `/mkdocs/config/typedoc.json`).

Jaguar0625 commented on mkdocs/scripts/ci/build.sh:
```diff
@@ -0,0 +1,8 @@
+set -ex
+
+# Run from the mkdocs folder
+export NO_MKDOCS_2_WARNING=1
```
what types of warnings are being suppressed?

segfaultxavi replied:
MkDocs is preparing to move to version 2 and will break compatibility.
Material, which is the MkDocs theme we use, will stop working if we move to MkDocs 2, so they print a very loud warning message:
 │  ⚠  Warning from the Material for MkDocs team
 │
 │  MkDocs 2.0, the underlying framework of Material for MkDocs,
 │  will introduce backward-incompatible changes, including:
 │
 │  × All plugins will stop working – the plugin system has been removed
 │  × All theme overrides will break – the theming system has been rewritten
 │  × No migration path exists – existing projects cannot be upgraded
 │  × Closed contribution model – community members can't report bugs
 │  × Currently unlicensed – unsuitable for production use
 │
 │  Our full analysis:
 │
 │  https://squidfunk.github.io/mkdocs-material/blog/2026/02/18/mkdocs-2.0/

This is the warning I am supressing with that envvar.

Jaguar0625 commented on mkdocs/scripts/ci/ezglossary.patch:
```diff
@@ -0,0 +1,12 @@
+--- a/src/mkdocs_ezglossary_plugin/plugin.py
```
i assume you opened bug for this in original repo?

segfaultxavi replied:
I opened a GitHub issue:
https://github.com/realtimeprojects/mkdocs-ezglossary/issues/26
And they did respond with a fix:
https://github.com/realtimeprojects/mkdocs-ezglossary/pull/27
I updated the dependency in commit 3b66ca2596 but for some reason the fix did not work for me.
But I just tested it again and seems to be working fine, so I will drop this patch.
Thanks!

Jaguar0625 commented on mkdocs/scripts/ci/lint_python.sh:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
you probably should update to use latest py linters after @Wayonb changes

segfaultxavi replied:
I didn't find any meaningful difference with the python linters used in sdk/python.
What should I update, @Wayonb?

Jaguar0625 commented on mkdocs/scripts/ci/test.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+echo "no tests for docs (for now)"
```
do you intend to eventually run some of your examples (in test_examples) ?

segfaultxavi replied:
Not anymore. I'll remove the "(for now)".
The linters already do a pretty good job at making sure the code compiles, and most tests would require funds.

Jaguar0625 commented on mkdocs/scripts/snippet_migration/extract_snippet_ranges.py:
```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
```
this was one time script, correct?

Jaguar0625 commented on mkdocs/scripts/deploy-gh-pages.sh:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
whats is intended purpose of this vs gh_pages_publish ?

segfaultxavi replied:
This script is my convenient way of building locally and publishing to GitHub pages.
If we are finally moving to GH building and publishing the docs on every commit, then we can move this script to the proper CI files like `ci/build.sh` and `ci/gh_pages_publish.sh`.
But I'll need help to do that.

Jaguar0625 replied:
ok, i will defer to @Wayonb on that

Jaguar0625 commented on mkdocs/scripts/gen_ref_pages_py.py:
```diff
@@ -0,0 +1,38 @@
+ignored_folders = mkdocs_gen_files.config['extra']['symbol']['py-sdk']['ignore-folders']
+
+root = Path(__file__).parent.parent.parent
+src = root / "sdk/python/symbolchain"
```
try not to mix " and ' in same file. we typically prefer '. we used to have linter rule but i think it was in deprecated plugin (cc: @Wayonb )

segfaultxavi replied:
Fixed!

Jaguar0625 commented on mkdocs/scripts/gen_ref_pages_ts.py:
```diff
@@ -0,0 +1,27 @@
+	if path.stem == "README" or path.stem == ".meta":
+		continue
+	if "nem" in path.parts:
+		continue
```
presumably you'll need to do something different for nem docs?

segfaultxavi replied:
Yes. In the NEM docs we skip the `symbol` folder :)

Jaguar0625 commented on mkdocs/scripts/hooks.py:
```diff
@@ -0,0 +1,513 @@
+						section[value] = current_section
+						counter += 1
+					elif isinstance(value, list):
+						walk(value, name)
```
are you intentionally ignoring result?

segfaultxavi replied:
`walk` writes directly onto the parent function's local variables, it does not returning anything.

Jaguar0625 commented on mkdocs/scripts/hooks.py:
```diff
@@ -0,0 +1,513 @@
+		meta, title = parse_page_header(text)
+
+		if "tutorial_level" not in meta:
+			continue
```
should this be required and error when not present?

segfaultxavi replied:
Good idea, but it's not trivial, since there are some pages which are not tutorials, and some tutorials which do not need the tag (like the Setup guide).
I could force all pages to have the metadata, and allow for `None`.
Or I could have a list of folders where pages must have the metadata.

All these options seemed dirty and more work than needed, given how we are not planning to add new tutorials for now.

Jaguar0625 commented on mkdocs/scripts/hooks.py:
```diff
@@ -0,0 +1,513 @@
+
+class ignoreRESTAnchors(logging.Filter):
+	def filter(self, record):
+		return not re.search(r'reference/rest/symbol.md.*does not contain an anchor', record.msg)
```
why this ?

segfaultxavi replied:
The REST reference guide is generated at run-time, therefore, any link to its anchors found at build-time produces a noisy warning.
This filter removes warnings due to missing anchors in links to the REST reference guide.

It's not ideal because it could silence true errors, but the dozens of false warnings were deemed more annoying.

Jaguar0625 replied:
👍 maybe add a comment to that effect

segfaultxavi replied:
Done!

Jaguar0625 commented on mkdocs/scripts/ptl.py:
```diff
@@ -0,0 +1,67 @@
+# Prepare Tasks List summary
```
is this integrated in docs or just used in reports external to docs ?

segfaultxavi replied:
Just used in reports that I no longer need, so I've removed this file.

Jaguar0625 commented on mkdocs/scripts/register_cats_lexer.py:
```diff
@@ -0,0 +1,9 @@
+	'CATS',           # display name
+	('cats',),        # aliases
+	('*.cats',),      # file extensions
+	('text/x-cats',)  # mimetypes
```
i assume you want to align the comments ?

segfaultxavi replied:
Most definitely :)

Jaguar0625 commented on mkdocs/scripts/ci/lint_python.sh:
```diff
@@ -0,0 +1,26 @@
+set -ex
+
+find . \( -name node_modules -o -name .venv \) -prune -o -type f -name "*.sh" -print0 | xargs -0 shellcheck
+find snippets \( -name node_modules -o -name .venv \) -prune -o -type f -name "*.py" -print0 | PYTHONPATH=. xargs -0 python3 -m isort \
+	--line-length 75 \
+	--indent "	" \
+	--multi-line 3 \
+	--check-only
+find snippets \( -name node_modules -o -name .venv \) -prune -o -type f -name "*.py" -print0 | PYTHONPATH=. xargs -0 python3 -m pycodestyle \
+	--config=.pycodestyle
+
+# Build a custom .pylintrc file based on the global one
+TMP_RC_FILE=/tmp/symbol-docs.pylintrc
+cp "$(git rev-parse --show-toplevel)/linters/python/.pylintrc" "$TMP_RC_FILE"
+{
+	# Allow lowercase "constants" (actually, top-level regular variables)
+	echo "const-rgx=(([A-Za-z_][A-Za-z0-9_]*)|(t_[A-Z0-9_]+)|(__.*__))$"
+	# Disable some warnings we accept for tutorial code
+	echo "disable=missing-docstring,broad-exception-caught,duplicate-code,use-maxsplit-arg,too-many-locals,too-many-branches,too-many-statements"
+	# Do not check these modules, as we do not install them to build the docs
+	echo "ignored-modules=web3,websockets"
+} >> $TMP_RC_FILE
+find snippets \( -name node_modules -o -name .venv \) -prune -o -type f -name "*.py" -print0 | PYTHONPATH=. xargs -0 python3 -m pylint \
+	--rcfile $TMP_RC_FILE
```
is fine to use modified rulesets for snippets in docs as we discussed before.

but you should be linting helper and build scripts with original rulesets. currently it looks like you have them excluded from _any_ linting.

segfaultxavi replied:
You're right, build scripts were not being linted. Fixed now.

Jaguar0625 reviewed (commented):
👍

Wayonb commented on .github/codecov.yaml:
```diff
@@ -58,6 +58,11 @@ flags:
       - client/rest
     carryforward: true
 
+  mkdocs:
```
Not sure if you need to configure code coverage for docs? If you are not running the sample code then I would remove.

If you want code coverage, then also define a project above ☝🏾

Wayonb commented on .github/dependabot.yaml:
Also need to add ``/mkdocs`` here to cover JavaScript dependencies.

Wayonb replied:
I think you forgot to add this 😄

segfaultxavi replied:
The review took longer than expected and I still had not pushed the changes. Done now 😛

Wayonb commented on jenkins/docker/ubuntu/docs.Dockerfile:
FYI: will probably also change to install the doxygen that ships with Ubuntu.  I see this failing in Jenkins, but want to make sure that's ok.

segfaultxavi replied:
I'm not sure what are you asking of me here.
Doxygen is only used to produce the Java docs, which are currently disabled.
So now is a good time to experiment with this :D

Wayonb commented on mkdocs/.eslintrc:
```diff
@@ -0,0 +1,45 @@
+extends:
+  - airbnb
+  - plugin:jsdoc/recommended-error
+  - ../linters/javascript/default.eslintrc
```
Should update to use ../linters/javascript/default.mjs and convert to ESLint flat config.
https://eslint.org/docs/latest/use/configure/migration-guide#migrate-your-config-file

segfaultxavi replied:
Done!

Wayonb reviewed (changes requested):
Look good👍🏾 
Have a few comments

segfaultxavi committed:
[mkdocs] fix: syntax highlight inline CATS snippets

So they look like block snippets.

segfaultxavi committed:
[mkdocs] fix: CATS textbook problems

segfaultxavi committed:
[mkdocs] fix: complete CATS lexer for syntax highlight

Previous one worked well enough for the examples, but was not complete.
This one has been generated from the Lark grammar.

segfaultxavi committed:
[mkdocs] fix: remove unnecessary code coverage check for docs

segfaultxavi committed:
[mkdocs] fix: warning about preloaded hover images in home page not being used

segfaultxavi committed:
[mkdocs] fix: 404 that works locally too

Previously it produced an endless redirection loop because the
redirection URL was relative.

segfaultxavi committed:
[mkdocs] fix: redirects and 404s

segfaultxavi committed:
[mkdocs] feat: remove root docs folder

All files needed there are created in a post-build step:
- Main root index redirect
- 404 redirect
- Redirects for specific files listed in the mkdocs config

segfaultxavi committed:
[mkdocs] fix: gh-pages deployment scripts

segfaultxavi committed:
[mkdocs] fix: generate required gh-pages static files

segfaultxavi committed:
[mkdocs] fix: move root folder files to templates

segfaultxavi committed:
[mkdocs] fix: align comments

segfaultxavi committed:
[mkdocs] fix: stop patching ezglossary

The plugin was fixed upstream and we already use the updated version.

segfaultxavi committed:
[mkdocs] fix: remove unused catch parameter

Keeps linter happy.

segfaultxavi committed:
[mkdocs] fix: update linter to eslint 10

segfaultxavi committed:
[mkdocs] fix: have dependabot checks docs JS deps

segfaultxavi committed:
[mkdocs] fix: remove unused scripts

They were just kep as reference.

segfaultxavi committed:
[mkdocs] fix: lint build scripts along with code snippets

THIS BREAKS LINTING of the build scripts.
Fixes come in the following commits.

segfaultxavi committed:
[mkdocs] fix: lint build scripts WIP

segfaultxavi committed:
[mkdocs] fix: lint build scripts WIP

segfaultxavi committed:
[mkdocs] fix: lint build scripts WIP

Wayonb commented on mkdocs/eslint.config.mjs:
```diff
@@ -0,0 +1,53 @@
+	{
+		...defaultConfig,
+		plugins: {
+			...defaultConfig.plugins,
```
Not really needed since ``...defaultConfig`` will include all, and ESLint should merge with any config after.

segfaultxavi committed:
[mkdocs] fix: remove unnecessary lint config

segfaultxavi committed:
[mkdocs] fix: add comments to the MkDocs custom hooks

This file has grown a lot and needed comments.
Might even need to be broken up in smaller pieces.

segfaultxavi committed:
[mkdocs] fix: clearer No Tests message

Jaguar0625 commented on mkdocs/scripts/typedoc_plugin.py:
```diff
@@ -0,0 +1,148 @@
+
+def is_typedoc_summary_file(dirpath, filename):
+	stem, extension = os.path.splitext(filename)
+	return (
```
extract `".md" == extension` since it's on both sides

segfaultxavi replied:
It was split in two because they are two different types of summary files, but it's no big deal. Fixed.

segfaultxavi committed:
[mkdocs] fix: simplify typedoc page filtering

zero4862 committed:
[mkdocs] docs: improve anchor positioning in tutorials (#2070)

Brings https://github.com/NemProject/nem/pull/920 to Symbol docs.

Jaguar0625 reviewed (approved):
👍

segfaultxavi committed:
[mkdocs] fix: remove false claim about typed descriptor tutorials

segfaultxavi committed:
[mkdocs] fix: lint error

pylint is getting confused by the `hooks` module and the `hooks` package (folder).

zero4862 committed:
[mkdocs] docs: style api reference (#2073)

Brings https://github.com/NemProject/nem/pull/922 to Symbol.

Wayonb reviewed (approved):
👍🏾
